client/app/scripts/utils/string-utils.js

```diff
@@ -4,43 +4,54 @@ import d3 from 'd3';
 
 const formatLargeValue = d3.format('s');
 
-const formatters = {
-  filesize(value) {
-    const obj = filesize(value, {output: 'object'});
-    return formatters.metric(obj.value, obj.suffix);
-  },
-
-  integer(value) {
-    if (value < 1100 && value >= 0) {
-      return Number(value).toFixed(0);
-    }
-    return formatLargeValue(value);
-  },
+function renderHtml(text, unit) {
+  return (
+    <span className="metric-formatted">
+      <span className="metric-value">{text}</span>
+      <span className="metric-unit">{unit}</span>
+    </span>
+  );
+}
 
-  number(value) {
-    if (value < 1100 && value >= 0) {
-      return Number(value).toFixed(2);
+
+function makeFormatters(renderFn) {
+  const formatters = {
+    filesize(value) {
+      const obj = filesize(value, {output: 'object', round: 1});
+      return renderFn(obj.value, obj.suffix);
+    },
+
+    integer(value) {
+      const intNumber = Number(value).toFixed(0);
+      if (value < 1100 && value >= 0) {
+        return intNumber;
+      }
+      return formatLargeValue(intNumber);
+    },
+
+    number(value) {
+      if (value < 1100 && value >= 0) {
+        return Number(value).toFixed(2);
+      }
+      return formatLargeValue(value);
+    },
+
+    percent(value) {
+      return renderFn(formatters.number(value), '%');
     }
-    return formatLargeValue(value);
-  },
-
-  percent(value) {
-    return formatters.metric(formatters.number(value), '%');
-  },
-
-  metric(text, unit) {
-    return (
-      <span className="metric-formatted">
-        <span className="metric-value">{text}</span>
-        <span className="metric-unit">{unit}</span>
-      </span>
-    );
-  }
-};
-
-export function formatMetric(value, opts) {
-  const formatter = opts && formatters[opts.format] ? opts.format : 'number';
-  return formatters[formatter](value);
+  };
+
+  return formatters;
+}
+
+
+const formatters = makeFormatters(renderHtml);
+const svgFormatters = makeFormatters((text, unit) => `${text}${unit}`);
+
+export function formatMetric(value, opts, svg) {
```

Comment:
Why didnt you export 2 formatMetric functions? You had to define 2 formatters (including the `svg` argument) anyway.